### PR TITLE
Add metrics

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

Central Dogma server lacks observability.

Modifications:

- Added `CentralDogma.meterRegistry()` which exposes a Micrometer
  `MeterRegistry` to users.
- Removed `CentralDogma.cacheStats()` because such information can be
  retrieved from `meterRegistry()`.
- Added `PluginContext.meterRegistry()` so that a `Plugin` can register
  a `Meter`.
- Added various meters:
  - System stats
  - Executor stats
  - Caffeine stats
  - ZooKeeper stats
  - HTTP API stats
- Exposed the metrics at `/monitor/metrics` so that Prometheus can scrape the
  metrics.

Result:

Observability

### CODE REVIEW

File: server/src/test/java/com/linecorp/centraldogma/server/MetricsTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 LINE Corporation
```
Comment (hyangtack) [inline server/src/test/java/com/linecorp/centraldogma/server/MetricsTest.java]: nit: 2019 :-)

File: server/src/main/java/com/linecorp/centraldogma/server/internal/storage/project/DefaultProjectManager.java [hunk 2]
```diff
@@ -35,24 +33,22 @@
...
     @Nullable
     private final RepositoryCache cache;
 
-    public DefaultProjectManager(File rootDir, Executor repositoryWorker, @Nullable String cacheSpec) {
+    public DefaultProjectManager(File rootDir, Executor repositoryWorker, MeterRegistry meterRegistry,
+                                 @Nullable String cacheSpec) {
         super(rootDir, Project.class);
         this.repositoryWorker = requireNonNull(repositoryWorker, "repositoryWorker");
-        cache = cacheSpec != null ? new RepositoryCache(cacheSpec) : null;
+        cache = cacheSpec != null ? new RepositoryCache(cacheSpec, meterRegistry) : null;
```
Comment (hyangtack) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/storage/project/DefaultProjectManager.java]: nit: requireNonNull(meterRegistry, "meterRegistry") ?

File: server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/RepositoryCache.java [hunk 3]
```diff
@@ -67,6 +70,8 @@ public RepositoryCache(String cacheSpec) {
                            logger.debug("Cache miss: {}", key);
                            return key.execute();
                        });
+
+        CaffeineCacheMetrics.monitor(meterRegistry, cache, "repository");
```
Comment (minwoox) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/RepositoryCache.java]: `requireNonNull(meterRegistry, "meterRegistry")`?

File: server/src/test/java/com/linecorp/centraldogma/server/internal/replication/ZooKeeperCommandExecutorTest.java [hunk 4]
```diff
@@ -250,6 +254,36 @@ public void stopWhileWaitingForInitialQuorum() throws Exception {
                                .hasMessageContaining("before joining");
     }
 
+    @Test
+    public void metrics() throws Exception {
+        final List<Replica> cluster = buildCluster(1, true /* start */,
+                                                   ZooKeeperCommandExecutorTest::newMockDelegate);
+        try {
+            final Map<String, Double> meters = MoreMeters.measureAll(cluster.get(0).meterRegistry);
+            assertThat(meters).containsKeys("executor.pool.size#value{name=zkCommandExecutor}",
```
Comment (hyangtack) [inline server/src/test/java/com/linecorp/centraldogma/server/internal/replication/ZooKeeperCommandExecutorTest.java]: How about checking time-related keys as well?

Comment (trustin) [reply server/src/test/java/com/linecorp/centraldogma/server/internal/replication/ZooKeeperCommandExecutorTest.java]: Sure.

**Comment by trustin** [comment]
Thanks a lot for reviewing. :-)

**Pull request merged**